content/oop_macro.md

```diff
@@ -175,28 +229,79 @@ macro class*(head: expr, body: stmt): stmt {.immediate.} =
   #           ObjectTy
   #             Empty
   #             OfInherit
-  #               Ident !"RootObj"
+  #               Ident !"TObject"
   #             Empty   <= We want to replace this
-  #   MethodDef
-  #   ...
-
-  result[0][0][0][2][0][2] = recList
-
+  type_decl[0][0][2][0][2] = recList
+  result.insert(0, type_decl)
+  
   # Lets inspect the human-readable version of the output
   # echo repr(result)
   # Output:
-  #  type
+  #  type 
   #    Animal = ref object of RootObj
   #      name: string
   #      age: int
   #
-  #  method vocalize(this: Animal): string =
+  #  method vocalize(this: Animal): string
+  #  method vocalizeAnimal(this: Animal): string   <= this is the magic for calling base class methods
+  #  method age_human_yrs(this: Animal): int
+  #  method age_human_yrsAnimal(this: Animal): int <= this is the magic for calling base class methods
```

Comment:
This isn't useful, [`procCall`](http://goran.krampe.se/2014/11/30/nim-and-oo-part-iv/) exists. It's quite new, and I'm not aware of it being documented anywhere though.